--- zorba-3.1/CMakeLists.txt.orig	2015-06-16 13:26:38.000000000 +0200
+++ zorba-3.1/CMakeLists.txt	2023-09-02 19:45:56.814184602 +0200
@@ -503,6 +503,10 @@ ENDIF (WIN32)
 
 # C++11 TR1 header location & namespace
 CHECK_CXX_SOURCE_COMPILES("
+  #include <type_traits>
+  int main() { std::is_convertible<int,int> x; }" ZORBA_NO_TR1)
+IF (NOT ZORBA_NO_TR1)
+CHECK_CXX_SOURCE_COMPILES("
   #include <tr1/type_traits>
   int main() { }" ZORBA_TR1_IN_TR1_SUBDIRECTORY)
 CHECK_CXX_SOURCE_COMPILES("
@@ -515,6 +519,7 @@ CHECK_CXX_SOURCE_COMPILES("
 IF (ZORBA_TR1_1 OR ZORBA_TR1_2)
   SET (ZORBA_TR1_NS_IS_STD_TR1 1)
 ENDIF (ZORBA_TR1_1 OR ZORBA_TR1_2)
+ENDIF (NOT ZORBA_NO_TR1)
 
 # C++11 langauge features
 CHECK_CXX_SOURCE_COMPILES("
